turbinia/lib/libcloudforensics.py

```diff
@@ -437,6 +437,66 @@ def get_or_create_analysis_vm(
     created = True
     return instance, created
 
+  def list_instance_by_label(self, labels_filter=''):
+    """Lists VMs in a project with on of the provided labels.
+
+    Args:
+      labels_filter: a dict of labels to find --> {'id': '123'}
```

Comment:
This being a dict is inconsistent with the default value in `labels_filter` (which is supposed to be a dict). In this case if labels_filter is not provided, a string will be passed to `__list_by_label` on which `items()` will be called, and bad things will happen.

Suggestion: Refactor `list_instance_by_label` and `list_disk_by_label` to have `labels_filter` be a mandatory argument (the function's names imply you're filtering on something so it makes sense to always pass some kind of filter IMO).

Reply:
oh, yes this should be changed for sure ! acknowledged and will be changed in the next commit

Reply:
Done